http/src/main/java/io/micronaut/http/annotation/CustomHttpMethod.java

```diff
@@ -38,24 +38,32 @@
 @HttpMethodMapping
 public @interface CustomHttpMethod {
     /**
-     * @return The URI of the GET route if not specified inferred from the method name and arguments
+     * @return The URI of the route
      */
     @AliasFor(annotation = HttpMethodMapping.class, member = "value")
     @AliasFor(annotation = UriMapping.class, member = "value")
     String value() default UriMapping.DEFAULT_URI;
 
     /**
+     * @return The URI of the route
+     */
+    @AliasFor(annotation = HttpMethodMapping.class, member = "value")
+    @AliasFor(annotation = UriMapping.class, member = "value")
+    String uri() default UriMapping.DEFAULT_URI;
+
+    /**
+     * Only to be used in the context of a server.
      *
-     * @return The name of the non-standard http method.
+     * @return The URIs of the route
      */
-    String method();
+    @AliasFor(annotation = HttpMethodMapping.class, member = "uris")
+    @AliasFor(annotation = UriMapping.class, member = "uris")
+    String[] uris() default {UriMapping.DEFAULT_URI};
```

Comment:
Do we need another member? Wouldn't it be ok to just change `value` and `uri` to return `String[]`? That way you could also do:

```
@Get("/foo", "/bar")
```